src/lib.rs

```diff
@@ -133,22 +135,37 @@ pub enum Error {
     Panicking(Box<Any + Send>),
 }
 
-impl Debug for Error {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self {
-            &Error::Finished => write!(f, "Finished"),
-            &Error::Waiting => write!(f, "Waiting"),
-            &Error::Panicked => write!(f, "Panicked"),
-            &Error::Panicking(ref err) => {
-                let msg = match err.downcast_ref::<&'static str>() {
+impl Error {
```

Comment:
Why not put these implementations into `impl error::Error for Error` block below?